envoy/http/filter.h

```diff
@@ -432,11 +433,16 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * @param modify_headers supplies an optional callback function that can modify the
    *                       response headers.
    * @param grpc_status the gRPC status code to override the httpToGrpcStatus mapping with.
+   * @param grpc_error_details the gRPC status error details. It will be seralized and encoded in
+   * grpc-status-details-bin header. Please be aware, inside `grpc_error_details`, only the
+   * `details` field will be effective and the `code`/`message` fields will be overwritten by the
+   * provided parameters.
    * @param details a string detailing why this local reply was sent.
    */
   virtual void sendLocalReply(Code response_code, absl::string_view body_text,
                               std::function<void(ResponseHeaderMap& headers)> modify_headers,
                               const absl::optional<Grpc::Status::GrpcStatus> grpc_status,
+                              std::unique_ptr<::google::rpc::Status> grpc_error_details,
```

Comment:
How about using default param value of `nullptr` so we don't have to change every caller of `sendLocalReply`? Is this acceptable Envoy style?

https://en.cppreference.com/w/cpp/language/default_arguments

Reply:
It is not recommended to have default arguments in virtual functions. See

https://stackoverflow.com/a/3533695
go/cstyle#Default_Arguments

Reply:
I see, good to know. How about function overload then? Otherwise it will be a lot of work for you to change all callers

Reply:
Two thoughts
1) I'm fine with optional default-to-null because refactoring adds a ton of churn.
2) If we opt to refactor rather than add a single field I'd be inclined to make the gRPC data its own optional struct so we can add to it more easily.
No strong preference between 1) and 2), but I do prefer we pick one of the two :-)